discovery-provider/src/queries/notifications.py

```diff
@@ -46,8 +57,124 @@ def get_owner_id(session, entity_type, entity_id):
     else:
         return None
 
+def get_cosign_remix_notifications(session, max_block_number, remix_tracks):
+    """
+    Get the notifications for remix tracks that are reposted/favorited by the parent remix author
+
+    Args:
+        session: (DB)
+        max_block_number: (int)
+        remix_tracks: (Array<{ }>)
+            'user_id'
+            'item_id'
+            const.notification_blocknumber
+            const.notification_timestamp
+            'item_owner_id'
+    
+    Returns:
+        Array of cosign notifications
+
+    """
+    if len(remix_tracks) == 0:
+        return []
+
+    remix_notifications = []
+    remix_track_ids = [r['item_id'] for r in remix_tracks]
+
+    # Query for all the parent tracks of the remix tracks 
+    tracks_subquery = (
+        session.query(Track)
+            .filter(
+                Track.is_unlisted == False,
+                Track.is_delete == False,
+                Track.blocknumber <= max_block_number
+            )
+            .distinct(Track.track_id)
```

Comment:
I think it might be better to just query by is_current? that'll enforce distinct as a side effect

Reply:
=1

Reply:
Well b/c we do the query with a max block number, the track might not represent the same metadata that we're trying to fetch. Thoughts @hareeshnagaraj about removing `max_block_number`?

Reply:
well since this is only for reposted/favorited, the tracks are most likely not within the same blocknumber. i think it's okay to not have that check here especially since this is a subquery for social actions and you have a list of track ids youre filtering with